baremetal/metal3cluster_manager_test.go

```diff
@@ -378,7 +370,7 @@ var _ = Describe("Metal3Cluster manager", func() {
 	)
 })
 
-func newBMClusterSetup(tc testCaseBMClusterManager) (*ClusterManager, error) {
```

Comment:
Is there any linter-related issue for this change? IMO returning an error is the better practice.

Reply:
Yes linter complains that this is always returning nill as an error, I agree its a good practice but the current code is not returning anything else. Hence I removed it.